woocommerce/wc-admin
# Refactor `MerchantEmailNotifications`

Fixes #8275

This PR refactors `MerchantEmailNotifications`.

- move src/Notes/MerchantEmailNotifications/NotificationEmail.php up into src/Notes (and keep it public)
- make /src/Notes/MerchantEmailNotificationss/MerchantEmailNotification.php internal
- rename `NotificationEmail` to `EmailNotification`

### Detailed test instructions:

- Check out this branch and create a `.zip` (by running `npm run test:zip`).
- Create a brand new site (using JN) and install and activate the `.zip`.
- Install the plugin [WP Mail Logging](https://wordpress.org/plugins/wp-mail-logging/). This plugin will log every sent mail.
- Turn on the merchant email notifications by going to `WooCommerce` > `Settings` > `Emails` and enabling the checkbox next to `Enable email insights`.
- Install the plugin [WooCommerce Admin Test Helper](https://github.com/woocommerce/woocommerce-admin-test-helper).
- Go to `Tools` > `WCA Test Helper` > `Tools` and press the `Run` button located in the same row of `Update WC installation timestamp` and set a date to 4 days before. We want a 4-day-old store.
- In the same tab run the `wc_admin_daily` job. 
- Go to `Tools` > `WP Mail Log` and verify that an email with the subject `Add your first product` has been sent.
- Press `View` (under the Message column). The email should be visible now in a popup.

![screenshot-popular-ladybug jurassic ninja-2022 02 11-16_59_48](https://user-images.githubusercontent.com/1314156/153661795-49644d0e-a30c-4aca-845b-ccf77fb9bacc.png)


- Verify that the Unit tests are running correctly as well.
